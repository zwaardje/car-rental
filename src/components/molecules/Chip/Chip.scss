$buildingUnit: 12px;
.chip {
    padding: $buildingUnit / 2;

    min-width: $buildingUnit * 5;
    display: flex;
    justify-content: flex-start;
    align-items: center;

    &.chip--primary {
        background-color: #ffffff;
        color: #444444;
        fill: #ffffff;
    }

    &.chip--active {
        background-color: #00b9bb;
        color: #ffffff;
        fill: #ffffff;
    }

    &.chip--secondary {
        background-color: #ffffff;
        color: #000000;

    }

    &__icon{
       display: flex;
       margin-right: 5px;
    }
}
