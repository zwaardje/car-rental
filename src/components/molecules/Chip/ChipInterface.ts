import { IconTypes } from '../../atoms';

export type ChipTypes = 'primary' | 'secondary' | 'active';

export interface Chip {
    type: ChipTypes;
    icon: IconTypes;
    label: string | number;
}
