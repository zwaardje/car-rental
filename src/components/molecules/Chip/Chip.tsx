import * as React from 'react';

import { Icon } from '../../atoms';
import { Chip as ChipInterface } from './ChipInterface';

import './Chip.scss';

export interface Props extends ChipInterface {}
export interface State {}

export default class Chip extends React.Component<Props, State> {
    static displayName = 'Chip';

    render() {
        const { icon, label, type } = this.props;

        return (
            <div className={`chip chip--${type}`}>
                <div className="chip__icon">
                    <Icon icon={icon} />
                </div>
                <span className="chip__label">
                    {label}
                </span>
            </div>
        );
    }
}
