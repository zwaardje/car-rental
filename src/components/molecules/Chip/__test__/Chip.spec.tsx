import * as React from 'react';
import * as renderer from 'react-test-renderer';

import Chip from '../Chip';

import {
    defaultProps,
} from '../__data__/Chip.mockdata';

describe('Chip', () => {

    it('renders defaultProps Chip correctly', () => {
        const props = { ... defaultProps };
        const tree = renderer.create(<Chip {... props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

});
