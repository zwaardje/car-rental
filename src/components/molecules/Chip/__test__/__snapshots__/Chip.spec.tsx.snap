// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip renders defaultProps Chip correctly 1`] = `
<div
  className="chip chip--primary"
>
  <div
    className="chip__icon"
  >
    <svg
      className="icon icon-check"
    >
      <use
        href="/assets/icons/iconset.svg#icon-check"
        xlinkHref="/assets/icons/iconset.svg#icon-check"
      />
    </svg>
  </div>
  <span
    className="chip__label"
  >
    3
  </span>
</div>
`;
